### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by hmrajas**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    <name>WSO2 Carbon - Analytics Publisher Library For File</name>
+    <url>http://wso2.org</url>
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.carbon</groupId>
+            <artifactId>org.wso2.carbon.logging</artifactId>
+            <version>${version.org.wso2.carbon}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon</groupId>
```
Comment (mohanvive) [inline pom.xml]: Why do we need carbon UI dependency for this extension.. ?

File: src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapter.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    /**
+     * Will be called after publishing is done, or when ConnectionUnavailableException is thrown.
+     */
+    public void disconnect() {
+        try {
+            for (String keyList : fileChannelMap.keySet()) {
+                fileChannelMap.get(keyList).close();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error occurred while close the FileChannel", e);
```
Comment (mohanvive) [inline src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapter.java]: Actually, above code needs to be corrected as below..

for (String keyList : fileChannelMap.keySet()) {
try{
            fileChannelMap.get(keyList).close();
} catch (IOException e){
log.error( ERROR MESSAGE, e);
}
           }

File: src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapterFactory.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        bufferSize.setHint(resourceBundle.getString(FileEventAdapterConstants.ADAPTER_BUFFER_HINT));
+        bufferSize.setDefaultValue(FileEventAdapterConstants.ADAPTER_BUFFER_SIZE_DEFAULT);
+        dynamicProperties.add(bufferSize);
+
+        Property fileName = new Property(FileEventAdapterConstants.ADAPTER_FILE_NAME);
+        fileName.setDisplayName(resourceBundle
+                .getString(FileEventAdapterConstants.ADAPTER_FILE_NAME));
+        fileName.setHint(resourceBundle.getString(FileEventAdapterConstants.ADAPTER_FILE_NAME_HINT));
+        dynamicProperties.add(fileName);
+
```
Comment (mohanvive) [inline src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapterFactory.java:89]: Make file Name as the first property..

File: src/main/java/org/wso2/carbon/extension/analytics/publisher/file/internal/util/FileEventAdapterConstants.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+package org.wso2.carbon.extension.analytics.publisher.file.internal.util;
+
+public class FileEventAdapterConstants {
+
+    private FileEventAdapterConstants() {
+    }
+
+    public static final String ADAPTER_TYPE = "File";
+
```
Comment (mohanvive) [inline src/main/java/org/wso2/carbon/extension/analytics/publisher/file/internal/util/FileEventAdapterConstants.java:27]: use simple characters.. "file"

File: src/main/java/org/wso2/carbon/extension/analytics/publisher/file/internal/util/FileEventAdapterConstants.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    public static final String ADAPTER_FILE_NAME = "filename";
+    public static final String ADAPTER_FILE_NAME_HINT = "filename.hint";
+
+    public static final String ADAPTER_BUFFER_SIZE = "buffersize";
+    public static final String ADAPTER_BUFFER_HINT = "buffersize.hint";
+
+    public static final String ADAPTER_BUFFER_SIZE_DEFAULT = "1024";
```
Comment (mohanvive) [inline src/main/java/org/wso2/carbon/extension/analytics/publisher/file/internal/util/FileEventAdapterConstants.java]: Use meaningful variable names please.. eg: ADAPTER_FILE_BUFFER_SIZE

File: src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapterFactory.java (hunk 3 above)
Comment (hmrajas1990) [inline src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapterFactory.java:89]: fixed

File: src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapter.java (hunk 2 above)
Comment (hmrajas1990) [inline src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapter.java]: fixed

File: src/main/java/org/wso2/carbon/extension/analytics/publisher/file/internal/util/FileEventAdapterConstants.java (hunk 5 above)
Comment (hmrajas1990) [inline src/main/java/org/wso2/carbon/extension/analytics/publisher/file/internal/util/FileEventAdapterConstants.java]: fixed

**Pull request merged**